android/app/src/main/java/fi/fmi/mobileweather/MaxForecastWidgetProvider.java

```diff
@@ -149,4 +157,26 @@ protected void setWidgetData(JSONArray announcementsJson, SharedPreferencesHelpe
 
         appWidgetManager.updateAppWidget(appWidgetId, widgetRemoteViews);
     }
+
+    private void setMaxWidgetSpecificColors(RemoteViews remoteViews, String theme) {
+        if (theme.equals(DARK) || theme.equals(GRADIENT)) {
+            remoteViews.setInt(R.id.timeTextView, "setTextColor", Color.WHITE);
```

Comment:
Instead of defining specific colors it would be better to add "styles" to colors.xml. You could have for example "timeTextColor" and white/dark color variants for it. Probably whole if-else-logic could be avoided if implemented like that. 

Easier to implement, easier to maintain if design changes.

Reply:
Öööh, one does not put styles to colors.xml. There is a separate styles.xml.

But setting colors is now simplified otherwise - also in BaseWidgetProvider.

Reply:
I meant that you can name colors like styles, instead of using color names like "blue".